### PR TITLE
Feature/deploy heroku button

### PR DESCRIPTION
**Pull request by iainkirkpatrick**
puts a 'deploy to heroku' button into the README, so that when the repo is cloned / forked to start a new project, a user can immediately deploy a live version to heroku

refer to https://github.com/root-systems/dogstack/issues/87

please squash commits as many were experimental 😅

### CODE REVIEW

**Comment by iainkirkpatrick** [comment]
also noting - the 'official' deployment (https://dogstack.herokuapp.com/) of this example app is out of date, once this PR is good then we should update that deployment too.

File: app.json [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+  "name": "dogstack example",
+  "description": "the example basic dogstack app",
+  "repository": "https://github.com/root-systems/dogstack-example",
+  "addons": [
+    "heroku-postgresql:hobby-dev",
+    "papertrail"
+  ],
+  "env": {
+    "APP_URL": {
+      "description": "the url that the app is running at (if you are deploying from the README, replace [app name] below with the app name you chose above)",
```
Comment (danalexilewis) [inline app.json:11]: @iainkirkpatrick It’s not clear what was called “[app name] above”

I’d also suggest adding some simple instructions to the reader file by the button stepping a new user through getting an app name and editing this app.json file before clicking the button.

Comment (iainkirkpatrick) [reply app.json:11]: @agentlewis do you mean “with the app name you chose above” is unclear? That refers to the form you get taken to when you click the button - is it unclear in the context of that form? 

The user shouldn’t have to edit anything before clicking the button - when they click and get to the form, they just make up a name for their app

Or am I misunderstanding you maybe?

Comment (danalexilewis) [reply app.json:11]: Ah sorry I didnt understand that the button took you to a form. I get it now.

Maybe just say the heroku app name. “Above” made me feel I missed something.

**Review by ahdinosaur — APPROVED** [review]

**Pull request merged**